lib/pbench/test/unit/server/query_apis/test_table_of_content.py

```diff
@@ -0,0 +1,206 @@
+                            "directory": "/default/sample",
+                            "mtime": "2021-05-21T24:00:00",
+                            "mode": "0o755",
+                            "name": "sample",
+                            "ancestor_path_elements": ["default"],
+                            "files": [
+                                {
+                                    "name": "result.txt",
+                                    "mtime": "2021-05-21T24:00:00",
+                                    "size": 0,
+                                    "mode": "0o644",
+                                    "type": "reg",
+                                },
+                                {
+                                    "name": "user-benchmark.cmd",
+                                    "mtime": "2021-05-21T24:00:00",
+                                    "size": 114,
+                                    "mode": "0o755",
+                                    "type": "reg",
+                                },
+                            ],
+                            "run_data_parent": "ece030b7b37ajsdgfa66s6a7a6be804",
+                            "authorization": {"owner": "1", "access": "private"},
+                            "@timestamp": "2021-05-21T24:00:00.141295",
+                        },
+                    }
+                ],
+            },
+        }
+        index = self.build_index_from_metadata()
+
+        # get_expected_status() expects to read username and access from the
+        # JSON client payload, however this API acquires that information
+        # from the Dataset. Construct a fake payload corresponding to the
+        # attach_dataset fixture.
+        auth_json = {"user": "drb", "access": "private"}
+        expected_status = self.get_expected_status(
+            auth_json, build_auth_header["header_param"]
+        )
+
+        response = query_api(
+            self.pbench_endpoint,
+            self.elastic_endpoint,
+            self.payload,
+            index,
+            expected_status,
+            json=response_payload,
+            status=HTTPStatus.OK,
+            headers=build_auth_header["header"],
+        )
+        if expected_status == HTTPStatus.OK:
+            res_json = response.json
+            expected_result = {
+                "directories": ["sample"],
+                "files": [
+                    {
+                        "name": "result.txt",
+                        "mtime": "2021-05-21T24:00:00",
+                        "size": 0,
+                        "mode": "0o644",
+                        "type": "reg",
+                    },
+                    {
+                        "name": "user-benchmark.cmd",
+                        "mtime": "2021-05-21T24:00:00",
+                        "size": 114,
+                        "mode": "0o755",
+                        "type": "reg",
+                    },
+                ],
+            }
```

Comment:
This is not the result that I would expect from the response at line 65:  that contains one hit which is (I think...) for a directory named `sample` which is a subdirectory of `default`.  Since the Pbench API request was ostensibly for the contents of `default`, the result should not include `result.txt` nor `user-benchmark.cmd` which are the contents of _`sample`_.  (In order for a result to contain both files and directories, I believe the ES response has to have multiple hits -- one for the files and one for each of the subdirectories.)

Reply:
oops.. seems like a blunder :D Now it seems I will have to change some query

Reply:
Yeah, if we can nail down what we want and how we want to address that, then I think it will be easier to construct tests for it.  😉

Once we're there, it would probably be good to have a series of tests here:
- an empty directory (which I think you have covered)
- a directory containing only files
- a directory containing only subdirectories
- a directory containing both files and subdirectories